### PR TITLE
added wait script for redis

### LINKED ISSUE #24: add wait script for redis
@svalee if u can

### PR DESCRIPTION
**Pull request by svalee**
this fixes #24

### CODE REVIEW

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
+    {%- endif %}
+    {%- if cookiecutter.use_redis == 'y' %}
+    redis = get_config(['-c', DEFAULT_CONFIG_PATH.as_posix()])['redis']
+    {%- endif %}
     while True:
+        {%- if cookiecutter.use_postgres == 'y' %}
         try:
             with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
                 sock.connect((postgres['host'], postgres['port']))
                 break
```
Comment (hzlmn) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py]: I postgres will be connected redis not be waited and vise versa

Comment (Arfey) [reply create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py]: @svalee use different `while` block

File: create_aio_app/template/hooks/post_gen_project.py [hunk 2]
```diff
@@ -8,20 +8,26 @@
 )
 
 
+def remove_database_redis_files() -> None:
```
Comment (Arfey) [inline create_aio_app/template/hooks/post_gen_project.py]: U use only one wait script, so remove only one file. And i think` remove_wait_script` is better name for this function

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py [hunk 3]
```diff
@@ -1,17 +1,30 @@
...
         except socket.error:
             time.sleep(0.1)
+        {%- endif %}
+        {%- if cookiecutter.use_redis == 'y' %}
+        try:
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+                sock.connect((redis['host'], redis['port']))
+                break
+        except socket.error:
+            time.sleep(0.1)
```
Comment (webknjaz) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py]: Please use an exponential backoff

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py [hunk 4]
```diff
@@ -1,17 +1,30 @@
-{%- if cookiecutter.use_postgres == 'y' %}
-
-import socket
+{%- if cookiecutter.use_postgres == 'y' or cookiecutter.use_redis == 'y' %}import socket
```
Comment (webknjaz) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py]: Rather than shipping empty module I'd use hooks to remove if unneeded.

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/common.py [hunk 5]
```diff
@@ -9,7 +9,7 @@
 
 
 PATH = pathlib.Path(__file__).parent.parent.parent
-settings_file = os.environ.get('SETTINGS_FILE', 'api.dev.yaml')
+settings_file = os.environ.get('SETTINGS_FILE', 'api.dev.yml')
```
Comment (webknjaz) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/common.py]: How is this related to awaiting redis?

File: create_aio_app/template/hooks/post_gen_project.py [hunk 6]
```diff
@@ -8,20 +8,26 @@
...
         os.remove(f'{file_name}')
 
-    for wait_file in WAIT_SERVICES_FILES:
-        os.remove(f'{{ cookiecutter.project_name }}/{wait_file}')
-
 
 def main() -> None:
-    if "{{ cookiecutter.use_postgres }}".lower() == "n":
+    without_postgres = "{{ cookiecutter.use_postgres }}".lower() == "n"
+    without_redis = "{{ cookiecutter.use_redis }}".lower() == 'n'
```
Comment (webknjaz) [inline create_aio_app/template/hooks/post_gen_project.py]: maybe use jinja2 filter? ` | lower`

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py [hunk 7]
```diff
@@ -1,17 +1,31 @@
...
+    {%- endif %}
+    {%- if cookiecutter.use_redis == 'y' %}
+    redis = config['redis']
+    while True:
+        try:
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+                sock.connect((redis['host'], redis['port']))
+                break
+        except socket.error:
+            time.sleep(0.1 + (random.randint(0, 100) / 1000))
```
Comment (hzlmn) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py]: From user perspective would be nice to have some logs to know why service not started and which dependent service not respond as well.

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py (hunk 4 above)
Comment (svalee) [reply create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py]: The module is removed if neither postgres, nor redis is used in the project

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py [hunk 8]
```diff
@@ -1,17 +1,35 @@
...
-import socket
+{%- if cookiecutter.use_postgres == 'y' or cookiecutter.use_redis == 'y' %}import socket
 import time
+import random
 
 from {{ cookiecutter.project_name }}.utils.common import get_config, DEFAULT_CONFIG_PATH
 
 if __name__ == '__main__':
-    postgres = get_config(['-c', DEFAULT_CONFIG_PATH.as_posix()])['postgres']
+    {%- if cookiecutter.use_postgres == 'y' or cookiecutter.use_redis == 'y' %}
```
Comment (Arfey) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py]: if we don't use postgres or redis we remove this file, so this `if` is not required.

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py [hunk 9]
```diff
@@ -1,17 +1,35 @@
-{%- if cookiecutter.use_postgres == 'y' %}
-
-import socket
+{%- if cookiecutter.use_postgres == 'y' or cookiecutter.use_redis == 'y' %}import socket
```
Comment (Arfey) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/wait_script.py]: `socket` is required, so remove` if`

**Comment by Arfey** [comment]
thx!
